fe/fe-core/src/main/java/com/starrocks/common/proc/FrontendsProcNode.java

```diff
@@ -74,16 +74,7 @@ public ProcResult fetchResult() {
     }
 
     public static void getFrontendsInfo(Catalog catalog, List<List<String>> infos) {
-        String masterIp = "";
-        int masterPort = -1;
-        try {
-            InetSocketAddress master = catalog.getHaProtocol().getLeader();
-            masterIp = master.getAddress().getHostAddress();
-            masterPort = master.getPort();
-        } catch (Exception e) {
-            // this may happen when majority of FOLLOWERS are down and no MASTER right now.
-            LOG.warn("failed to get leader: {}", e.getMessage());
-        }
+        String masterIp = Catalog.getCurrentCatalog().getMasterIp();
```

Comment:
Is this ip equals with Leader's IP all the time?
Is it possible that getMasterIp return null?

Reply:
Catalog.getCurrentCatalog().getMasterIp() will return the real masterIp if catalog is ready, or fe will never provide query service.